Wikipedia/Code/PreviewWebViewContainer.swift

```diff
@@ -10,70 +10,79 @@ import WMF
     func wmf_showAlert(forTappedAnchorHref href: String)
 }
 
-@objcMembers class PreviewWebViewContainer: UIView, WKNavigationDelegate, WKScriptMessageHandler {
+class PreviewWebViewContainer: UIView, WKNavigationDelegate, WKScriptMessageHandler, Themeable {
     weak var externalLinksOpenerDelegate: WMFOpenExternalLinkDelegate?
-    var webView: WKWebView?
+    var theme: Theme = .standard
     @IBOutlet weak var previewSectionLanguageInfoDelegate: WMFPreviewSectionLanguageInfoDelegate!
     @IBOutlet weak var previewAnchorTapAlertDelegate: WMFPreviewAnchorTapAlertDelegate!
 
     func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
-        guard message.name == "anchorClicked", let messageDict = message.body as? [String: Any], let href = messageDict["href"] as? String else {
+        guard let href = href(from: message) else {
             return
         }
         previewAnchorTapAlertDelegate.wmf_showAlert(forTappedAnchorHref: href)
     }
     
-    func configuration() -> WKWebViewConfiguration {
-        let userContentController = WKUserContentController()
-        
-        let langInfo: MWLanguageInfo? = previewSectionLanguageInfoDelegate.wmf_editedSectionLanguageInfo()
-        let uidir = UIApplication.shared.wmf_isRTL ? "rtl" : "ltr"
-        
-        var earlyJavascriptTransforms: String? = nil
-        if let code = langInfo?.code, let dir = langInfo?.dir {
-            earlyJavascriptTransforms = """
-            document.onclick = function() {\
-            event.preventDefault();\
-            if (event.target.tagName == 'A'){\
-            var href = event.target.getAttribute( 'href' );\
-            window.webkit.messageHandlers.anchorClicked.postMessage({ 'href': href });\
-            }\
-            };\
-            window.wmf.utilities.setLanguage('\(code)', '\(dir)', '\(uidir)');
-            """
+    private func href(from message: WKScriptMessage) -> String? {
+        guard message.name == "anchorClicked", let messageDict = message.body as? [String: Any], let href = messageDict["href"] as? String else {
+            return nil
         }
+        return href
+    }
+    
+    private func configuration() -> WKWebViewConfiguration {
+        let controller = WKUserContentController()
+        var earlyJSTransforms = ""
+        if let langInfo = previewSectionLanguageInfoDelegate.wmf_editedSectionLanguageInfo() {
+            earlyJSTransforms = earlyJSTransformsString(for: langInfo, isRTL: UIApplication.shared.wmf_isRTL)
+        }
+        controller.addUserScript(WKUserScript(source: earlyJSTransforms, injectionTime: .atDocumentEnd, forMainFrameOnly: true))
+        controller.addUserScript(WKUserScript(source: "window.wmf.themes.classifyElements(document)", injectionTime: .atDocumentEnd, forMainFrameOnly: true))
+        controller.add(WeakScriptMessageDelegate(delegate: self), name: "anchorClicked")
```

Comment:
The link handling can likely be handled by setting the `webView`'s `navigationDelegate` instead of adding a click listener

Reply:
Haha good catch!

Reply:
Fixed!